### ISSUE TITLE
bump deltalake to version 0.7 and update scan_delta and scan_read function

### ISSUE THREAD

**Issue by dominikpeter** [pola-rs/polars#6752]
### Problem description

I suggest updating `deltalake` to the latest version and update scan_delta and scan_read to take advantage of the pickable `DeltaStorageHandler`.  This would enable delta-rs native object store support. Also, the current code would be easier.

Internally, I just do:

```
def scan_delta2(
    table_uri: str,
    version: int | None = None,
    storage_options: dict[str, Any] | None = None,
    delta_table_options: dict[str, Any] | None = None,
    pyarrow_options: dict[str, Any] | None = None,
) -> LazyFrame:

    dt = DeltaTable(table_uri, version=version, storage_options=storage_options, **delta_table_options or {})
   
    pa_ds = dt.to_pyarrow_dataset(**pyarrow_options or {})

    return pl.scan_ds(pa_ds)
```

@chitralverma what do you think?

**Comment by chitralverma** [comment]
okay I will take this up

**Comment by chitralverma** [comment]
PR for this minor change will be ready today.

pinning deltalake version to the latest as they made some fixes to object_store related things in 0.8.0 and 0.8.1
